### ISSUE TITLE
Intra-doc links cannot link to associated methods on type aliases.

### ISSUE THREAD

**Issue by jonhoo** [rust-lang/rust#86120]
I tried this code:

```rust
#[warn(rustdoc::all)]

pub struct Foo<C>(C);

impl<C> Foo<C> {
    pub fn new() -> Self {
        todo!()
    }
}

/// [`Bar::new`]
pub type Bar<C> = Foo<C>;
```

I expected to see this happen: `Bar::new` should be a link to `Foo::new`.

Instead, this happened: `Bar::new` was not turned into a link, and the compiler gave this warning:

```
warning: unresolved link to `Bar::new`
  --> src/lib.rs:11:6
   |
11 | /// [`Bar::new`]
   |      ^^^^^^^^^^ the type alias `Bar` has no associated item named `new`
   |
   = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: 1 warning emitted
```

I'm guessing that this isn't implemented because the forwarding may not always be obvious such as if there are multiple `fn new` for `Foo` for different values of `C`. But it'd be nice if it did work when there aren't and the choice is "clear".

EDIT: Though on second thought, that should also be an issue for
```rust
/// [`Foo::new`]
```
on `Foo`, so I feel like if that works this should also work.

### Meta

`rustc --version --verbose`:
```
rustc 1.52.1 (9bc8c42bb 2021-05-09)
binary: rustc
commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
commit-date: 2021-05-09
host: x86_64-apple-darwin
release: 1.52.1
LLVM version: 12.0.0
```

The same issue is present on nightly.

**Comment by jonhoo** [comment]
This is a simplified variant of #85960.

**Comment by inquisitivecrystal** [comment]
@rusbot label +A-intra-doc-links +T-rustdoc

**Comment by LeSeulArtichaut** [comment]
This can give some pretty surprising results:
```rs
pub struct Foo;

/// You should really try [`Self::bar`]
pub type Bar = Foo;

impl Bar {
    pub fn bar() {}
}
```
fails with
```
warning: unresolved link to `Bar::bar`
 --> test.rs:3:28
  |
3 | /// You should really try [`Self::bar`]
  |                            ^^^^^^^^^^^ the type alias `Bar` has no associated item named `bar`
  |
  = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: 1 warning emitted
```

**Comment by jyn514** [comment]
@LeSeulArtichaut this was fixed by #86334, right?

**Comment by LeSeulArtichaut** [comment]
If we consider that the link not going to the type alias page is not a bug, this is fixed
